### ISSUE TITLE
App lease renewer task failed

### ISSUE THREAD

**Issue by ThomasArdal** [Azure/azure-functions-durable-extension#2345]
### Description

A durable function of mine has been running fine for months. Last night I got the following message logged thousands of times:

```
myappname-applease: App lease renewer task failed. AppLeaseId: d7734cdf-0000-0000-0000-000000000000 Exception: System.ObjectDisposedException: The CancellationTokenSource has been disposed. at DurableTask.AzureStorage.Partitioning.AppLeaseManager.LeaseRenewer(CancellationToken cancellationToken)
```

I'm wondering why this happens. I could find any instances of this created as an issue why I'm creating it. I'm not sure if it's a bug or not. And I'm not sure if this is something that I can do anything to avoid going forward. Does anyone know what this means and if there is something to be done here other than just ignoring this error going forward?

### App Details

```xml
<TargetFramework>net6.0</TargetFramework>
<AzureFunctionsVersion>v4</AzureFunctionsVersion>
```

```xml
<PackageReference Include="Azure.Storage.Blobs" Version="12.14.1" />
<PackageReference Include="Elmah.Io.Client.Extensions.SourceCode" Version="4.1.22-pre" />
<PackageReference Include="Elmah.Io.Extensions.Logging" Version="4.2.33-pre" />
<PackageReference Include="Microsoft.Azure.Functions.Extensions" Version="1.1.0" />
<PackageReference Include="Microsoft.Azure.WebJobs.Extensions.DurableTask" Version="2.8.1" />
<PackageReference Include="Microsoft.Azure.WebJobs.Extensions.ServiceBus" Version="4.3.0" />
<PackageReference Include="Microsoft.NET.Sdk.Functions" Version="4.1.1" />
<PackageReference Include="Microsoft.Extensions.Http" Version="6.0.0" />
<PackageReference Include="Microsoft.Azure.WebJobs.Extensions.DurableTask.Analyzers" Version="0.5.0">
	<PrivateAssets>all</PrivateAssets>
</PackageReference>
```

This happened around `2022-12-27T21:06:35.0546745Z` in the `West US` region on worker `RD00155DE9AF0B`.

**Comment by davidmrdavid** [comment]
Hi @ThomasArdal. Although I'm not an expert on the app-lease renewal process, I'm fairly confident that these leases may fail occasionally without affecting the overall health of your app. So if your application is running correctly, I think you can safely ignore these logs.

For further details and any corrections, I'll defer to @amdeel

**Comment by christoph-spitze** [comment]
@ThomasArdal did you ever find a solution for this issue? We are having the same problem and our durable function is sending lots of errors into our AppInishts instance.

**Comment by froeschel** [comment]
@ThomasArdal id you ever find a solution for this issue? We are having the same problem and our durable function is sending lots of errors into our AppInishts instance.

**Comment by ThomasArdal** [comment]
@froeschel No, I don't think I did. I believe I ignored those issues (together with all of the other errors generated by Functions that are "expected" 👎)

**Comment by froeschel** [comment]
@ThomasArdal thanks for the quick reply.  Yes, it looks like the function is executing correctly, but it is spamming errors. Looks like I also will need to find a way of ignoring these errors in our AppInsights.

**Comment by ThomasArdal** [comment]
@davidmrdavid The problem here is that this message is logged with a severity of `Error`. I could just ignore all errors from the logger name (like `DurableTask.Core`) but I'm not sure I'd want to ignore everything from this logger.

**Comment by davidmrdavid** [comment]
@ThomasArdal: Thanks for clarifying. Hmm, that's interesting. Can you confirm for certain that you're seeing `DurableTask.Core` as the source of this log, instead of `DurableTask.AzureStorage`? That would help narrow down our options here. Thanks!

**Comment by ThomasArdal** [comment]
@davidmrdavid I'll try to remove the ignore filter to monitor if this happens again. I'm afraid I no longer have the original error so I don't know if the source were `DurableTask.Core` or `DurableTask.AzureStorage`. When looking at the error I think that the source is probably `DurableTask.AzureStorage`. My point here is that I don't want to ignore ALL errors from `DurableTask.AzureStorage`. Or maybe I will. Depending on any if any error coming from this source is actually something that I an do anything about 😊

**Comment by davidmrdavid** [comment]
Thanks for clarifying, that makes sense.
From a quick scan of the code, it appears this error is being thrown [here](https://github.com/Azure/durabletask/blob/3dd624ea067b6a3b2da9c1ae5210e5348b2da22b/src/DurableTask.AzureStorage/Partitioning/AppLeaseManager.cs#L427-L433) and I see it is definitely an error-level log. Also, that line tells me it's in `AzureStorage` so I don't think it's necessary anymore for you to remove your filter, but of course you're welcomed to if you would really like to confirm.

I need to discuss with the team to understand our options here. At a glance, I understand why this is logged as an error; so I'm not immediately clear on the best course of action. I'll defer to the members of the team I just tagged here to get their thoughts.

**Comment by froeschel** [comment]
@davidmrdavid @ThomasArdal I was a bit more pragmatic here. I filtered out all errors from ``DurableTask.AzureStorage`` in my alert rule. On our site I'm sure that the source was ``DurableTask.AzureStorage``.

**Comment by ThomasArdal** [comment]
Thank you so much for looking into this, David. I totally understand why the functions package wants to log errors when they happen. The challenge here is that we as users of the packages that also set up logging get a lot of internal errors logged from the functions packages. And often it is errors that we cannot do anything about. So, the answer is typical that this is expected behavior and that the error can be ignored. It's hard to know which errors to take seriously and which ones to ignore. I don't have any good solutions, though.
